test/uk/gov/hmrc/cataloguefrontend/CatalogueControllerSpec.scala

```diff
@@ -140,10 +142,11 @@ class CatalogueControllerSpec extends WordSpec with MockitoSugar {
     implicit val headerCarrrier: HeaderCarrier = HeaderCarrier()
 
     val deploymentsService = mock[DeploymentsService]
-    val configuration      = mock[Configuration]
+    val configuration      = mock[ServicesConfig]
+    val mcc = mock[MessagesControllerComponents]
 
-    when(configuration.getString("microservice.services.user-management.profileBaseUrl"))
-      .thenReturn(Some("profile-base-url"))
+    when(configuration.getConfString("microservice.services.user-management.profileBaseUrl", ""))
+      .thenReturn("profile-base-url")
 
     val controller = new CatalogueController(
```

Comment:
There is a general question of how we would like to use templates: inject them or reference them through object defs? If we'd be injecting them, probably we can mock them in specs' setups and stop using GuiceOneAppPerSuite and what maybe even more important, controllers tests will be purely verifying logic coded in the endpoints not trying to find something suitable in a template to use assertions like: `html includes something` which doesn't seems to be very robust and easy to understand. I know this approach comes with a cost of doing a bit more changes to the existing specs but maybe we could suggest it to the teams as an option for existing services and a prefered approach for new services?